File: 283. Move Zeroes/moveZeroes.js
/**
 * @param {number[]} nums
 * @return {void} Do not return anything, modify nums in-place instead.
 */
const moveZeroes = (nums) => {
  let nonZeroIndex = 0;

  for (let i = 0; i < nums.length; i++) {
    if (nums[i] !== 0) {
      nums[nonZeroIndex] = nums[i];
      nonZeroIndex++;
    }
  }

  for (let j = nonZeroIndex; j < nums.length; j++) {
    nums[j] = 0;
  }

  return nums;
}

const nums = [0, 1, 0, 3, 12];
console.log('moveZeroes: ', moveZeroes(nums));

File: 283. Move Zeroes/moveZeroes.py
class Solution:
  def moveZeroes(self, nums: list[int]) -> None:
    """
    Do not return anything, modify nums in-place instead.
    """
    nonZeroIndex = 0

    for i in range(len(nums)):
      if nums[i] != 0:
        nums[nonZeroIndex] = nums[i]
        nonZeroIndex += 1

    for i in range(nonZeroIndex, len(nums)):
      nums[i] = 0

nums = [0, 1, 0, 3, 12]
solution = Solution()
print('moveZeroes: ', solution.moveZeroes(nums))